src/test/test-copy.c

```diff
@@ -249,8 +249,8 @@ TEST(copy_tree) {
         assert_se(stat(unixsockp, &st) >= 0);
         assert_se(S_ISSOCK(st.st_mode));
 
-        assert_se(copy_tree(original_dir, copy_dir, UID_INVALID, GID_INVALID, COPY_REFLINK) < 0);
-        assert_se(copy_tree("/tmp/inexistent/foo/bar/fsdoi", copy_dir, UID_INVALID, GID_INVALID, COPY_REFLINK) < 0);
+        assert_se(copy_tree(original_dir, copy_dir, UID_INVALID, GID_INVALID, COPY_REFLINK, NULL) < 0);
+        assert_se(copy_tree("/tmp/inexistent/foo/bar/fsdoi", copy_dir, UID_INVALID, GID_INVALID, COPY_REFLINK, NULL) < 0);
 
```

Comment:
please add a unit test to cover the deny list

Reply:
@DaanDeMeyer this one is unaddressed, it would be good to have a unit test case here that covers the case where denylist isn't NULL, as sanity check - doesn't need to be overly complex, just to ensure nothing obvious breaks

Reply:
Added a unit test